app/controllers/signatures_controller.rb

```diff
@@ -19,6 +19,10 @@ def create
     end
   end
 
+  def signed
+    @petition = Petition.find(params[:petition_id])
```

Comment:
What if we moved the ``signed`` action to ``/petitions/<id>/signed`` ?  The signature isn't important to the page we show.   Or we should do as otto suggests and make sure the ``<petition_id>`` and ``<signature_id>`` match up.